src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloadsTestBase.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Testing;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.PreferStreamAsyncMemoryOverloads,
+    Microsoft.NetCore.Analyzers.Runtime.PreferStreamAsyncMemoryOverloadsFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreferStreamAsyncMemoryOverloadsTestBase
+    {
+        #region Helpers
+
+        protected static async Task VerifyAnalyzerAsync50(string source, params DiagnosticResult[] expected)
+        {
+            var test = new VerifyCS.Test
+            {
+                TestCode = source,
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp31,
```

Comment:
The memory extensions APIs were introduced in .NET Core 2.1. Do we have to test the analyzer in all versions, or is any version above 2.1 sufficient?

https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.readasync?view=netcore-2.1
https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.writeasync?view=netcore-2.1